### PR TITLE
Add support for supplementary result files

### PR DESCRIPTION
**Pull request by julianbreiteneicher**
This PR enables experiments to create supplementary
result files in addition to the main result file.

The `CommitReport.get_supplementary_file_name` method
can be used to generate a supplementary report file.
Multiple supplementary files per revision are also
possible.
They can be differentiated by the methods `info_type`
parameter.

The experiment can then use the generated filenames
to create these files and store additional information
or results of the experiment (e.g. the experiments duration
time).

Contrary to the "normal" result files, these supplementary
files do not indictate a success or failure of the experiment.
They are intended to provide additional information about
successful experiment runs.

This means that if an experiment fails, supplementary files
that have already been created should be deleted.
This can easily be done by the experiment by passing a
list of files to the experiments `PEErrorHandler`.

If the experiment fails and the error handler is called,
it will automatically delete all files of the passed list.

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/revisions.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4231983&bid=14466489)

File: varats/data/revisions.py [hunk 1]
```diff
@@ -97,3 +136,34 @@ def get_tagged_revisions(project_name: str, result_file_type: MetaReport
                                   str(newest_res_file))))
 
     return revisions
+
+
+def get_supplementary_result_files(project_name: str,
+                                   result_file_type: MetaReport,
+                                   revision: tp.Optional[str] = None
+                                   ) -> tp.List[tp.Tuple[Path, str, str]]:
```
Comment (vulder) [inline varats/data/revisions.py]: Should we maybe allow the user to pass a info_type? So we could filter an only return supp files that are needed.

**Review by vulder — CHANGES_REQUESTED** [review]
Other than my remark, looks good :+1: 

PS: could you please resolve the merge conflicts

File: varats/data/revisions.py (hunk 1 above)
Comment (julianbreiteneicher) [reply varats/data/revisions.py]: @vulder: Done

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/revisions.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4231983&bid=14466489)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/revisions.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4231983&bid=14466489)

**Review by vulder — APPROVED** [review]
lgtm :)
I'll merge when the conflicts are resolved

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/revisions.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4231983&bid=14466489)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/revisions.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4231983&bid=14466489)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/revisions.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4231983&bid=14466489)

**Comment by julianbreiteneicher** [comment]
> lgtm :)
> I'll merge when the conflicts are resolved

done.

**Pull request merged**